### DISCUSSION TITLE
Can app dependencies being injected into `startup` events via `Depends()`?

### DISCUSSION THREAD

**Discussion by br-follow** [fastapi/fastapi#8736] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.on_event("startup")
def access_settings(settings: Settings = Depends(get_settings)):
    print(settings.setting_name)
```

### Description

Is is possible to access app dependencies in a startup event?

When I try this in my application, I get

```
ERROR:    Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 621, in lifespan
    async with self.lifespan_context(app):
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 518, in __aenter__
    await self._router.startup()
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 598, in startup
    await handler()
  File "/code/./app/api/server.py", ...
AttributeError: 'Depends' object has no attribute 'setting_name'
```

### Operating System

Linux

### FastAPI Version

0.70.1

### Python Version

3.10.1

**Comment by chbndrhnns** [comment, marked as answer]
No, the dependency injection feature depends on the request context which is only available with an incoming request.
